ftpcontroller.py

```diff
@@ -72,7 +75,13 @@ def readData(self, command, argument, dataSocket):
 
   # TODO Handle missing file
   def sendData(self, argument, dataSocket):
-    sourceFile = open(argument, 'rb')
+    try:
+      sourceFile = open(argument, 'rb')
+    except FileNotFoundError:
+      dataSocket.close()
+      self.dumpResponseBuffer()
+      self.appendToBuffer('File \'{}\' not found within current directory.'.format(argument))
```

Comment:
We aren’t sending the ABOR command? Also we should return the same message that we are adding to the buffer.